### ISSUE TITLE
option to report unreachable code as warning or error

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ty#3321]
currently ty only reports unreachable code as an IDE hint, but in most cases unreachable code is a mistake, so it should be reported as a diagnostic instead:

```py
if False:
    print(1) # hint: Code is always unreachable
```

these hints are extremely easy to miss, especially if relying on ty in CI to catch mistakes.

ideally, all diagnostics that can be reported as an IDE hint should have an option to be reported as any other diagnostic level such as warning or error. see my comment in https://github.com/astral-sh/ruff/pull/23305#discussion_r2908269477 and https://github.com/astral-sh/ty/issues/784#issuecomment-3368973563 for previous discussion
